docs/translations/translation-loading.md

```diff
@@ -0,0 +1,331 @@
+
+	$handle_src      = explode( '/build/', $wp_scripts->registered[ $handle ]->src );
+	$handle_filename = $handle_src[1];
+	$locale          = determine_locale();
+	$lang_dir        = WP_LANG_DIR . '/plugins';
+
+	// Translations are always based on the unminified filename.
+	if ( substr( $handle_filename, -7 ) === '.min.js' ) {
+		$handle_filename = substr( $handle_filename, 0, -7 ) . '.js';
+	}
+
+	// WordPress 5.0 uses md5 hashes of file paths to associate translation
+	// JSON files with the file they should be included for. This is an md5
+	// of 'packages/woocommerce-blocks/build/FILENAME.js'.
+	$core_path_md5     = md5( 'packages/woocommerce-blocks/build/' . $handle_filename );
+	$core_json_file    = $lang_dir . '/woocommerce-' . $locale . '-' . $core_path_md5 . '.json';
+	$json_translations = is_file( $core_json_file ) && is_readable( $core_json_file ) ? file_get_contents( $core_json_file ) : false; // phpcs:ignore
+
+	if ( ! $json_translations ) {
+		return $translations;
+	}
+
+	// Rather than short circuit pre_load_script_translations, we will output
+	// core translations using an inline script. This will allow us to continue
+	// to load feature-plugin translations which may exist as well.
+	$output = <<<JS
+	( function( domain, translations ) {
+		var localeData = translations.locale_data[ domain ] || translations.locale_data.messages;
+		localeData[""].domain = domain;
+		wp.i18n.setLocaleData( localeData, domain );
+	} )( "{$domain}", {$json_translations} );
+JS;
+
+	if ( empty( $wp_scripts->done ) ) {
+		// If we hadn't printed any script into the page, let's enqueue the translations.
+		// phpcs:ignore WordPress.WP.EnqueuedResourceParameters.NoExplicitVersion
+		wp_register_script( $handle_filename, '', array( 'wp-i18n' ), false, true );
+		wp_enqueue_script( $handle_filename );
+		wp_add_inline_script(
+			$handle_filename,
+			$output
+		);
+	} else {
+		// If we have already printed scripts into the page, there is a chance that
+		// scripts have finished being printed. That means that if we enqueued them here,
+		// they would never be printed. Instead of enqueuing, then, let's print directly
+		// the script tag.
+		printf( "<script type='text/javascript'>\n%s\n</script>\n", $output ); // phpcs:ignore
+	}
+
+	// Finally, short circuit the pre_load_script_translations hook by returning
+	// the translation JSON from the feature plugin, if it exists so this hook
+	// does not run again for the current handle.
+	$path_md5     = md5( 'build/' . $handle_filename );
+	$json_file    = $lang_dir . '/' . $domain . '-' . $locale . '-' . $path_md5 . '.json';
+	$translations = is_file( $json_file ) && is_readable( $json_file ) ? file_get_contents( $json_file ) : false; // phpcs:ignore
+
+	if ( $translations ) {
+		return $translations;
+	}
+
+	// Return valid empty Jed locale.
+	return '{ "locale_data": { "messages": { "": {} } } }';
+}
+
+add_filter( 'pre_load_script_translations', 'woocommerce_blocks_get_i18n_data_json', 10, 4 );
```

Comment:
This filter and corresponding callback were refactored in #6478. Do you think we should update this example to prevent confusion?

Reply:
I've just updated that code example, as suggested by you.